frontend/app/routes/protected/person-case/primary-docs.tsx

```diff
@@ -46,37 +57,64 @@ export async function action({ context, request }: Route.ActionArgs) {
   const t = await getFixedT(request, handle.i18nNamespace);
 
   const formData = await request.formData();
+  const action = formData.get('action');
+  const documentTypeAvailable = formData.get('documentType') != null;
 
-  if (formData.get('action') === 'back') {
-    throw i18nRedirect('routes/protected/person-case/privacy-statement.tsx', request); //TODO: change it to redirect to file="routes/protected/person-case/request-details.tsx"
-  }
+  switch (action) {
+    case 'back': {
+      throw i18nRedirect('routes/protected/person-case/request-details.tsx', request);
+    }
 
-  // submit action
-  const schema = v.object({
-    currentStatusInCanada: v.pipe(
-      v.string(t('protected:primary-identity-document.current-status-in-canada.required')),
-      v.trim(),
-      v.nonEmpty(t('protected:primary-identity-document.current-status-in-canada.required')),
-      v.literal(
-        'canadian-citizen-born-outside-canada',
-        t('protected:primary-identity-document.current-status-in-canada.invalid'),
-      ),
-    ),
-  });
-
-  const input = { currentStatusInCanada: formData.get('currentStatusInCanada') as string };
-  const parsedDataResult = v.safeParse(schema, input, { lang });
-
-  if (!parsedDataResult.success) {
-    return data({ errors: v.flatten<typeof schema>(parsedDataResult.issues).nested }, { status: 400 });
-  }
+    case 'next': {
+      const schema = v.intersect([
+        v.object({
+          currentStatusInCanada: v.pipe(
+            v.string(t('protected:primary-identity-document.current-status-in-canada.required')),
+            v.trim(),
+            v.nonEmpty(t('protected:primary-identity-document.current-status-in-canada.required')),
+            v.picklist(VALID_CURRENT_STATUS, t('protected:primary-identity-document.current-status-in-canada.invalid')),
+          ),
+        }),
+        v.variant(
+          'documentTypeAvailable',
+          [
+            v.object({
+              documentTypeAvailable: v.literal(false),
+              documentType: v.optional(v.string()),
+            }),
+            v.object({
+              documentTypeAvailable: v.literal(true),
+              documentType: v.pipe(v.picklist(VALID_DOCTYPE, t('protected:primary-identity-document.document-type.required'))),
+            }),
+          ],
+          t('protected:primary-identity-document.document-type.required'),
+        ),
+      ]) satisfies v.GenericSchema<PrimaryDocumentsSessionData>;
```

Comment:
Validation schema can be simplified like this. `currentStatusInCanada` and `documentType` are both required field and must be a valid string value from the `picklist`. There is nothing bad to validate `documentType` even though it's not displayed on the screen. The error for that field will not be displayed.

```suggestion
      const schema = v.object({
          currentStatusInCanada: v.picklist(VALID_CURRENT_STATUS, t('protected:primary-identity-document.current-status-in-canada.required')),
            documentType: v.picklist(VALID_DOCTYPE, t('protected:primary-identity-document.document-type.required')),
            }) satisfies v.GenericSchema<PrimaryDocumentsSessionData>;
```

Reply:
I tested it locally, and since 'documentType' might not be visible, returning an error when it's hidden causes the error to immediately appear when the field becomes visible. It feels odd. Here's an improved validation approach:

```ts
const schema = v.pipe(
  v.object({
    currentStatusInCanada: v.picklist(
      VALID_CURRENT_STATUS,
      t('protected:primary-identity-document.current-status-in-canada.required'),
    ),
    documentType: v.string(),
  }),
  // Perform an additional check on 'documentType' only when currentStatusInCanada is valid.
  // Otherwise, it incorrectly triggers an error even when 'documentType' is not visible.
  v.forward(
    v.partialCheck(
      [['currentStatusInCanada'], ['documentType']],
      (input) => VALID_DOCTYPE.includes(input.documentType),
      t('protected:primary-identity-document.document-type.required'),
    ),
    ['documentType'],
  ),
) satisfies v.GenericSchema<PrimaryDocumentsSessionData>;

const input = {
  currentStatusInCanada: String(formData.get('currentStatusInCanada')),
  documentType: String(formData.get('documentType')),
} satisfies Partial<v.InferInput<typeof schema>>;
```

Reply:
updated